smarts/env/gymnasium/wrappers/metric/utils.py

```diff
@@ -95,3 +95,55 @@ def multiply(value: Union[int, float], multiplier: Union[int, float]) -> float:
         float: Value x Multiplier
     """
     return float(value * multiplier)
+
+
+class SlidingWindow:
+    """A sliding window which moves to the right by accepting new elements. The
+    maximum value within the sliding window can be queried at anytime by calling
+    the max() method. 
+    """
+    def __init__(self, size:int):
+        """
+        Args:
+            size (int): Size of the sliding window.
+        """
+        self._values = deque(maxlen=size)
+        self._max_candidates = deque(maxlen=size)
+        self._size = size
+        self._time = -1
+
+    def move(self, x:Union[int,float]):
+        """Moves the sliding window one step to the right by appending the new
+        element x and discarding the oldest element on the left.
+
+        Args:
+            x (Union[int,float]): New element input to the sliding window.
+        """
+        self._time += 1
+
+        # When values deque is full, remove head element of max_candidates deque 
+        # if it matches head element of values deque.
+        if len(self._values) == self._size:
+            if self._values[0][0] == self._max_candidates[0][0]:
+                self._max_candidates.popleft()
+        # Append x to values deque.
+        self._values.append((self._time,x))
+
+		# Remove elements from max_candidates deque's tail which are less than x.
```

Comment:
There appears to be excess whitespace.
```suggestion
	# Remove elements from max_candidates deque's tail which are less than x.
```